mapbox/encoding.py

```diff
@@ -53,7 +61,7 @@ def encode_waypoints(features, min_limit=None, max_limit=None, precision=6):
     return a string encoded in waypoint-style used by certain mapbox APIs
     ("lon,lat" pairs separated by ";")
     """
-    coords = ['{lon:.{p}f},{lat:.{p}f}'.format(lon=lon, lat=lat, p=precision)
+    coords = ['{lon},{lat}'.format(lon=round(lon, precision), lat=round(lat, precision))
```

Comment:
Putting the waypoints in the URL as the Matrix API does (which I'm 👎  on to be honest) exposes some bugginess in printing with precision. Rounding and then printing works better and is backwards compatible.